tests/src/webhooks/utils/handleUserFeatures.test.ts

```diff
@@ -0,0 +1,163 @@
+  let usersTiersRepository: UsersTiersRepository;
+  let productsRepository: ProductsRepository;
+  let bit2MeService: Bit2MeService;
+  let defaultProps: HandleUserFeaturesProps;
+
+  beforeEach(() => {
+    tiersRepository = testFactory.getTiersRepository();
+    usersRepository = testFactory.getUsersRepositoryForTest();
+    usersRepository = testFactory.getUsersRepositoryForTest();
+    displayBillingRepository = {} as DisplayBillingRepository;
+    couponsRepository = testFactory.getCouponsRepositoryForTest();
+    usersCouponsRepository = testFactory.getUsersCouponsRepositoryForTest();
+    usersTiersRepository = testFactory.getUsersTiersRepository();
+    productsRepository = testFactory.getProductsRepositoryForTest();
+    bit2MeService = new Bit2MeService(config, axios);
+    paymentService = new PaymentService(
+      new Stripe(config.STRIPE_SECRET_KEY, { apiVersion: '2024-04-10' }),
+      productsRepository,
+      bit2MeService,
+    );
+    usersService = new UsersService(
+      usersRepository,
+      paymentService,
+      displayBillingRepository,
+      couponsRepository,
+      usersCouponsRepository,
+      config,
+      axios,
+    );
+    tiersService = new TiersService(usersService, paymentService, tiersRepository, usersTiersRepository, config);
+
+    defaultProps = {
+      user: mockedUser,
+      purchasedItem: mockedPurchasedItem,
+      paymentService,
+      customer: mockedCustomer,
+      tiersService,
+    };
+  });
+
+  it('when the product does not exists, then an error indicating so is thrown', async () => {
+    const tierNotFoundError = new TierNotFoundError('Tier not found');
+    jest.spyOn(tiersService, 'getTierProductsByProductsId').mockRejectedValue(tierNotFoundError);
+
+    await expect(handleUserFeatures(defaultProps)).rejects.toThrow(tierNotFoundError);
+  });
+
+  it('When the user does not have tiers, then it should insert a new tier', async () => {
+    const tierNotFoundError = new TierNotFoundError('Tier not found');
+    jest.spyOn(tiersService, 'getTierProductsByProductsId').mockResolvedValue(mockedTier);
+    jest.spyOn(tiersService, 'getTiersProductsByUserId').mockRejectedValue(tierNotFoundError);
+    const spyInsert = jest.spyOn(tiersService, 'insertTierToUser');
+    const spyUpdate = jest.spyOn(tiersService, 'updateTierToUser');
+    const spyApplyTier = jest.spyOn(tiersService, 'applyTier').mockResolvedValue();
+
+    await handleUserFeatures(defaultProps);
+
+    expect(spyInsert).toHaveBeenCalledTimes(1);
+    expect(spyInsert).toHaveBeenCalledWith(mockedUser.id, mockedTier.id);
+    expect(spyApplyTier).toHaveBeenCalledWith(
+      mockedUser,
+      mockedCustomer,
+      mockedPurchasedItem,
+      (mockedPurchasedItem.price?.product as Stripe.Product).id,
+    );
+    expect(spyUpdate).not.toHaveBeenCalled();
+  });
+
+  it('when the user has existing tiers and the second invoice has a product that is not mapped, then it should stop', async () => {
```

Comment:
Be more specific about what means "stopping". I understand that it means that is does not apply any tier, hence, the question is: how are you going to handle a case where a user does not have a tier but somewhat this handler has been called? That seems like a non-expected path that should be errored so Stripe reports a failed webhook on its UI.